### PR TITLE
Support additional attributes and secondary indexes

### PR DESCRIPTION
**Pull request by Jamie-BitFlight**
## what
* Added the ability to add additional attributes, and secondary indexes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,7 +94,18 @@ module "dynamodb_table" {
...
 | `enable_autoscaler`             | `true`       | Flag to enable/disable DynamoDB autoscaling                                    | No       |
+| `dynamodb_attributes`           | `[]`         | List of maps, that describe extra dynamodb attributes                          | No       |
+| `global_secondary_index_map`    | `[]`         | List of maps, that describes additional secondary index properties             | No       |
+
+
+## A note about dynamodb attributes
+Only define attributes on the table object that are going to be used as:
+
+* Table hash key or range key
+* LSI or GSI hash key or range key
```
Comment (osterman) [inline README.md]: Add new line after bullets

File: README.md [hunk 2]
```diff
@@ -47,7 +94,18 @@ module "dynamodb_table" {
 | `autoscale_min_write_capacity`  | `5`          | DynamoDB autoscaling min write capacity                                        | No       |
 | `autoscale_max_write_capacity`  | `20`         | DynamoDB autoscaling max write capacity                                        | No       |
 | `enable_autoscaler`             | `true`       | Flag to enable/disable DynamoDB autoscaling                                    | No       |
+| `dynamodb_attributes`           | `[]`         | List of maps, that describe extra dynamodb attributes                          | No       |
+| `global_secondary_index_map`    | `[]`         | List of maps, that describes additional secondary index properties             | No       |
+
+
+## A note about dynamodb attributes
```
Comment (osterman) [inline README.md]: Use proper capitalization for `DynamoDB`

File: README.md [hunk 3]
```diff
@@ -47,7 +94,18 @@ module "dynamodb_table" {
 | `autoscale_min_write_capacity`  | `5`          | DynamoDB autoscaling min write capacity                                        | No       |
 | `autoscale_max_write_capacity`  | `20`         | DynamoDB autoscaling max write capacity                                        | No       |
 | `enable_autoscaler`             | `true`       | Flag to enable/disable DynamoDB autoscaling                                    | No       |
+| `dynamodb_attributes`           | `[]`         | List of maps, that describe extra dynamodb attributes                          | No       |
```
Comment (osterman) [inline README.md]: `...extra DynamoDB attributes...`

**Comment by Jamie-BitFlight** [comment]
Will address.

**Comment by Jamie-BitFlight** [comment]
Great :) 
Good luck on the other ones!

**Comment by osterman** [comment]
Thanks @ankysh!

**Pull request merged**